### PR TITLE
[4.0]Beef up formbehavior

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes
Introducing 2 new shortcuts:
```php
// Load the default behaviours for singular form
JHtml::_('formbehavior.singular');

// Load the default behaviours for plural form
JHtml::_('formbehavior.plural');
```
Purpose: DRY

Also clean up some scripts

### Testing Instructions
Nothing breaks in the admin area

### Documentation Changes Required

### CODE REVIEW

**Comment by wilsonge** [comment]
What's the advantage of this over having a single method for this helper and additionally calling multiselect in the "plural" view

File: administrator/components/com_categories/views/category/tmpl/edit.php [hunk 1]
```diff
@@ -12,10 +12,12 @@
 // Include the component HTML helpers.
 JHtml::addIncludePath(JPATH_COMPONENT . '/helpers/html');
 
-JHtml::_('behavior.formvalidator');
-JHtml::_('behavior.keepalive');
```
Comment (C-Lodder) [inline administrator/components/com_categories/views/category/tmpl/edit.php]: `JHtml::_('behavior.keepalive');` you're removing this?

**Comment by Bakual** [comment]
Why load `behavior.core` with the new methods? Do we need it other than for `behavior.formvalidator` which already takes care of its dependancies itself?
Also why load the tabstate, tooltips and keepalive? If I don't need those in my form, they should not be loaded. Looking at your diif we haven't loaded them in many places before so I question the need for always loading them.

I agree with @wilsonge that it looks a bit pointless to have two methods that do the same except for the added multiselect. If anything, it could be a switch (eg `$multiselect = false`) in single method.

**Comment by dgt41** [comment]
@Bakual about 
- core.js: you are right
- tooltips: well right now there are already loaded in the index.php (which is wrong), also load them in the singular form is wrong, the right place is in every field. For list views is needed

The actual implementation might not be the perfect one, but the idea of having one call for each of the different views (singular, plural) makes some sense to me, not only it's DRYier but also is easier for new devs to pick up...
As always that's my personal view and it might not reflect the projects aims 😀

**Comment by mbabker** [comment]
The only thing DRY about it is not having a chain of several JHtml calls.  I don't see the need for two separate methods which are loading the same scripts.  A flag is more than enough on one method.

**Pull request closed without merging**